### ISSUE TITLE
UI Refactor Bug Megathread

### ISSUE THREAD

**Issue by Mirino97** [space-wizards/space-station-14#11949]
## Description
I will be compiling most reported UI issues, cropped up from the latest UI rework, here. I'll be pasting links to videos/images related instead of embedding to save some space. Some/most of these are probably already fixed, but this helps @Jezithyr and @DrSmugleaf track better and make sure they don't miss any.

Thanks to all the people who reported and helped us catalog these issues, your reports are invaluable!

- [x] HUD Themes stopped working (Gets set to Modernized, and after changing the selection, nothing happens)
- [X] CoolDown timer animation can get stuck in the hand slot if you drop the item used to invoke said animation, clearing it only after picking up another item. (https://cdn.discordapp.com/attachments/1029734309470412821/1030904989943615620/2022-10-15_19-09-35.mp4) #12048 should fix it.
- [x] Speech Bubbles can get hidden behind the game window (or Viewport, I think it's called) (#11937)
- [x] DoAfter timer for stripping (and a few other actions) can run for longer than the action takes to complete (https://cdn.discordapp.com/attachments/1029734309470412821/1031055627600019536/Space_Station_14_2022-10-16_00-52-07.mp4)
- [x] The Actions bar "layer" selector (the little numbers under the bar that represent in which bar you currently are) can come up out of order (https://cdn.discordapp.com/attachments/1029734309470412821/1030593672535875644/2022-10-14_22-32-37.mp4)
- [x] Clicking on actions sometimes require spamming to work (using your buttons doesn't)  (https://cdn.discordapp.com/attachments/1029734309470412821/1030222887195525150/2022-10-13_21-59-01.mp4)
- [x] Voting pop up is invisible (sorry, no image. Invisible. Get it? Hah!)
- [x] Wake up action doesn't appear after sleeping (https://cdn.discordapp.com/attachments/1029734309470412821/1029786410892410911/unknown.png)(https://cdn.discordapp.com/attachments/1029734309470412821/1029734893791481907/unknown.png)
- [x] Action Icon updating is broken (#12042).
- [X] Targeted actions no longer work (#12036). Should be fixed by #12046
- [x] Construction menu button no longer works #12033
- [x] Some action are added to the hotbar twice (e.g., when wearing magboots) https://cdn.discordapp.com/attachments/1029734309470412821/1030185047061450812/unknown.png)
- [x]  #12158 
- [x] Some shaders appear off-center when using oldhcat (not strictly a ui refactor's fault). https://discord.com/channels/310555209753690112/1031917007895007233 or #12177

Possibly already fixed:
- [ ] The chat-window sometimes steals focus from the dev console, rendering it unusable. I can't manage to reliably cause this, but it effectively renders it seems to require a restart in order to fix.
- [ ] Sometimes the entity spawn menu doesn't update when searching until you start scrolling.
- [ ] Stretch Viewport option doesn't stretch (https://cdn.discordapp.com/attachments/790656972801572905/1029753339459686491/unknown.png)
- [ ] Ghost Roles button shows up very small and with no text (https://cdn.discordapp.com/attachments/790656972801572905/1029752305869930496/unknown.png)
- [ ] Contents of Objects Tab on Admin Panel can't be right clicked

**Comment by keronshb** [comment]
Action charges don't get set anymore. You can either try setting them from the action, or the function .SetCharges() and they'll breakpoint with the values but will still act like they're infinite use.

**Comment by Streaky-Haddock** [comment]
Hotkeys and binds don't work on the lobby. Lobby chat is 'wiped' from the previous round but still shows new chat afte that. However clicking filters on and off 'refreshes' the chat log and gets it to show previous convos again.

However pressing 'T' to open the menu, tab to cycle channel and even binds like F1, F7 etc do not work. Things behave normally once the round begins.

**Comment by metalgearsloth** [comment]
Clicking on alerts doesn't work (e.g. click to stop pulling, click to toggle internals).

**Comment by Willhelm53** [comment]
Do these bugged right-click-names belong here? Both captured in-game on Wizard servers as a regular player with no admin/debugging view.

![Capture](https://user-images.githubusercontent.com/97707302/197071661-39f2f17d-de66-4604-97cb-ccf4a2e61e3f.PNG)
![Capture2](https://user-images.githubusercontent.com/97707302/197071668-f004cbe3-7309-4032-be63-9deb863a6577.PNG)

**Comment by Jezithyr** [comment]
> Do these bugged right-click-names belong here? Both captured in-game on Wizard servers as a regular player with no admin/debugging view.
> 
> ![Capture](https://user-images.githubusercontent.com/97707302/197071661-39f2f17d-de66-4604-97cb-ccf4a2e61e3f.PNG) ![Capture2](https://user-images.githubusercontent.com/97707302/197071668-f004cbe3-7309-4032-be63-9deb863a6577.PNG)

That's a localization bug, it's not related to the UI-Refactor. Just create a new issue for that and someone should be able to fix it! This won't require any code to fix :)

**Comment by TemporalOroboros** [comment]
> Clicking on alerts doesn't work (e.g. click to stop pulling, click to toggle internals).
> 
> Edit: Okay now it works so IDK what happened with the UI click recognition.

It seems to be broken specifically in the separated layout. I found that toggling back to the default UI layout let me unbuckle using it but as long as I was in the separated layout mode it was nonfunctional.

**Comment by Elijahrane** [comment]
https://github.com/space-wizards/space-station-14/issues/13029
Getting disabled from lack of charges or status effects doesn't work

**Comment by eoineoineoin** [comment]
> > Clicking on alerts doesn't work (e.g. click to stop pulling, click to toggle internals).
> > Edit: Okay now it works so IDK what happened with the UI click recognition.
> 
> It seems to be broken specifically in the separated layout. I found that toggling back to the default UI layout let me unbuckle using it but as long as I was in the separated layout mode it was nonfunctional.

I just tried to repro this and saw the opposite. It worked in the "Seperated" layout (my default) but not the "Default" layout. So, it seems to be something that's going wrong specifically when you change that setting in-game.

**Comment by deltanedas** [comment]
funny thing where scroll is interpolated from 0 to whatever it is upon research computer "autosyncing" im not sure what the term is

![scroll.webm](https://user-images.githubusercontent.com/39013340/211151599-24c3ee46-3579-48c3-99f5-66198a804a32.webm)

**Comment by h3half** [comment]
> funny thing where scroll is interpolated from 0 to whatever it is upon research computer "autosyncing" im not sure what the term is
> 
> ![scroll.webm](https://user-images.githubusercontent.com/39013340/211151599-24c3ee46-3579-48c3-99f5-66198a804a32.webm)

This is easily reproduced by placing a `base R&D point source`, opening an R&D Console, and scrolling down. You can also manually trigger the scroll action by clicking Sync. Super annoying!

**Comment by ElectroJr** [comment]
I'm fairly sure this issue can be closed now. Any remaining UI bugs should just have their own issues, and likely aren't directly related to the UI Refactor from 2+ years ago
